File: Python/BOJ_2747_피보나치 수.py
import sys

def fibo(n):
    if n < 2:
        return n
    
    cache = [0 for _ in range(n + 1)]
    cache[1] = 1
    
    for i in range(2, n + 1):
        cache[i] = cache[i - 2] + cache[i - 1]
    
    return cache[n]

num = int(sys.stdin.readline())
print(fibo(num))

File: C/BOJ_2747_피보나치 수.c
#include <stdio.h>

int fibo(n)
{
	int arr[50];
	arr[0] = 1;
	arr[1] = 1;
	for (int i = 2; i < 50; i++) arr[i] = arr[i - 2] + arr[i - 1];
	return arr[n - 1];
}

int main()
{
	int num;
	scanf("%d", &num);
	printf("%d", fibo(num));
	return 0;
}